caramel/config.py

```diff
@@ -71,3 +91,29 @@ def _get_config_value(
             env_var,
         )
     return result
+
+
+def get_log_level(argument_level, root=None, env=None):
+    """Calculates the highest verbosity(here inverted) from the argument,
+    environment and root, capping it to between 0-3, returning the log level"""
+    if env is None:
+        env = os.environ
+    env_level = int(env.get("CARAMEL_LOG_LEVEL", 0))
+
+    if root is None:
+        root = logging.getLogger()
+    root_level = LOG_LEVEL.index(root.level) if root.level in LOG_LEVEL else 0
+
+    verbosity = max(argument_level, env_level, root_level)
+    verbosity = min(verbosity, len(LOG_LEVEL) - 1)
+    log_level = LOG_LEVEL[verbosity]
+    return log_level
+
+
+def configure_log_level(arguments: argparse.Namespace, root=None):
```

Comment:
`root` isn't  "root logger" here, should be named `logger`  as when you use it, you pass it in a logger you wish to configure.

"root logger" is just the fallback.